controlplane/kubeadm/internal/workload_cluster_coredns.go

```diff
@@ -125,28 +125,33 @@ func (w *Workload) UpdateCoreDNS(ctx context.Context, kcp *controlplanev1.Kubead
 		return err
 	}
 
-	// Return early if the from/to image is the same.
-	if info.FromImage == info.ToImage {
-		return nil
-	}
+	// Only update the KubeadmConfigMap and CoreDNSCoreFile if the image changed.
+	if info.FromImage != info.ToImage {
+		// Validate the image tag.
+		if err := validateCoreDNSImageTag(info.FromImageTag, info.ToImageTag); err != nil {
+			return errors.Wrapf(err, "failed to validate CoreDNS")
+		}
 
-	// Validate the image tag.
-	if err := validateCoreDNSImageTag(info.FromImageTag, info.ToImageTag); err != nil {
-		return errors.Wrapf(err, "failed to validate CoreDNS")
+		// Perform the upgrade.
+		if err := w.updateCoreDNSImageInfoInKubeadmConfigMap(ctx, &clusterConfig.DNS, version); err != nil {
+			return err
+		}
+		if err := w.updateCoreDNSCorefile(ctx, info); err != nil {
+			return err
+		}
 	}
 
-	// Perform the upgrade.
-	if err := w.updateCoreDNSImageInfoInKubeadmConfigMap(ctx, &clusterConfig.DNS, version); err != nil {
-		return err
-	}
-	if err := w.updateCoreDNSCorefile(ctx, info); err != nil {
-		return err
-	}
+	// Update the cluster role independent of image change. Kubernetes may get updated
```

Comment:
What about moving this up L128, before all the changes that depends on the image changing

Reply:
I did not want to change the order of actions. But I'd be also ok with that :-) I think as long as it is still before a potential deployment change this should be fine.

Reply:
Yup +1 before is fine for me

Reply:
done, and also resolved the test issues.